pdfgen.c

```diff
@@ -437,13 +438,13 @@ static int pdf_set_err(struct pdf_doc *doc, int errval, const char *buffer,
     /* Make sure we're properly terminated */
     if (doc->errstr[len] != '\n')
         doc->errstr[len] = '\n';
-    doc->errstr[len] = '\0';
+    doc->errstr[len + 1] = '\0';
```

Comment:
I think the issue here is actually the two lines above should refer to doc->errstr[len - 1], and the nul termination should remain the same. The sizeof(doc->errstr) - 2 should be sizeof(doc->errstr) - 1 as well. This function is a great example of crappy string handling in C.

Reply:
Feel free to just drop this one and I'll try and sort it out if you want.

Reply:
Dropped.